test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Routing/Conventions/PropertyRoutingConventionTests.cs

```diff
@@ -235,5 +235,67 @@ public void SelectAction_ReturnsNull_IfPatchToCollectionProperty()
             Assert.Null(selectedAction);
             Assert.Empty(SelectActionHelper.GetRouteData(request).Values);
         }
+
+        [Fact]
+        public void SelectAction_ReturnsNull_IfPostToNonCollectionProperty()
+        {
+            // Arrange
+            CustomersModelWithInheritance model = new CustomersModelWithInheritance();
+            ODataPath odataPath = new DefaultODataPathHandler().Parse(model.Model, _serviceRoot, "VipCustomer/Address");
+            var request = RequestFactory.Create(HttpMethod.Post, "http://localhost/");
+            var actionMap = SelectActionHelper.CreateActionMap("PostToAddress");
+
+            // Act
+            string selectedAction = SelectActionHelper.SelectAction(new PropertyRoutingConvention(), odataPath, request, actionMap);
+
+            // Assert
+            Assert.Null(selectedAction);
+            Assert.Empty(SelectActionHelper.GetRouteData(request).Values);
+        }
+
+        [Theory]
+        [InlineData("Get", "Get")]
+        [InlineData("Put", "PutTo")]
+        [InlineData("Delete", "DeleteTo")]
+        [InlineData("Post", "PostTo")]
+        public void SelectAction_OnComplexCollection_ReturnsTheActionName(string httpMethod, string prefix)
+        {
+            // Arrange
+            CustomersModelWithInheritance model = new CustomersModelWithInheritance();
+            ODataPath odataPath = new DefaultODataPathHandler().Parse(model.Model, _serviceRoot, "Customers(7)/OtherAccounts");
+            var request = RequestFactory.Create(new HttpMethod(httpMethod), "http://localhost/");
+            var actionMap = SelectActionHelper.CreateActionMap(prefix + "OtherAccountsFromCustomer");
+
+            // Act
+            string selectedAction = SelectActionHelper.SelectAction(new PropertyRoutingConvention(), odataPath, request, actionMap);
+
+            // Assert
+            Assert.NotNull(selectedAction);
+            Assert.Equal(prefix + "OtherAccountsFromCustomer", selectedAction);
+            Assert.Single(SelectActionHelper.GetRouteData(request).Values);
+            Assert.Equal(7, SelectActionHelper.GetRouteData(request).Values["key"]);
+        }
+
+        [Theory]
+        [InlineData("Get", "Get")]
+        [InlineData("Put", "PutTo")]
```

Comment:
>"Put", "PutTo")] [](start = 20, length = 16)

Why "Complex" test case has "Delete", enum doesn't?

Reply:
+1: I assume we support Delete here too?

---
In reply to: [223519938](https://github.com/OData/WebApi/pull/1628#discussion_r223519938) [](ancestors = 223519938)

Reply:
EnumCollections do support delete but the property needs to be nullable for that, which is not the case with this property.